packages/application-generic/src/usecases/compile-email-template/compile-email-template.usecase.ts

```diff
@@ -149,7 +149,7 @@ export class CompileEmailTemplate {
   public static addPreheader(content: string): string {
     // "&nbsp;&zwnj;&nbsp;&zwnj;" is needed to spacing away the rest of the email from the preheader area in email clients
     return content?.replace(
-      /<body[^>]*>/g,
+      /<body[^>]*>/,
```

Comment:
can you try this regex `/<body\b[^>]*>/i` and see if it passes the code scanning.

Reply:
Both suggestions are still vulnerable.

Reply:
why is it vulnerable ?

Reply:
Unfortunately it is also vulnerable to DoS attacks that is the problem we are trying to solve with the original RegExp:
https://github.com/novuhq/novu/security/code-scanning/26
There are multiple tools to validate the safety of RegExp. For example: https://devina.io/redos-checker.

Reply:
but is not all the RegExp is vulnerable to DoS attack?

Reply:
No, not them all are. The dangerous patterns are described here, inside the `Evil Regex` section:
https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS

Reply:
I see, thanks for this info :fire: